app/views/pages/items.js

```diff
@@ -70,6 +70,13 @@ let ItemsViewController = React.createClass({
     }
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    if (this.getParams().id !== this.state.product.id) {
+      ProductActions.init(this.getParams().id);
```

Comment:
I looked at the blame on this method because I was thinking the same thing... but I think it might have been somewhere else. Anyway, every time you switch products `ProductActions.init` must be called so that pusher events and item listeners get wired up correctly.